example/lib/main.dart

```diff
@@ -30,13 +30,27 @@ class _MyAppState extends State<MyApp> {
   ably.Realtime _realtime;
   ably.Rest _rest;
   ably.ConnectionState _realtimeConnectionState;
+  ably.ChannelState _realtimeChannelState;
+  List<StreamSubscription<ably.ConnectionStateChange>> connectionStateChangeSubscriptions;
+  StreamSubscription<ably.ChannelStateChange> channelStateChangeSubscription;
 
   @override
   void initState() {
     super.initState();
     initPlatformState();
   }
 
+  @override
+  void dispose() {
+    // This dispose would not effective in this example, but subscriptions must
```

Comment:
typo "not be effective" - but, also, I think this comment needs expansion as you really need to explain _why_ it would not be effective.

Reply:
addressed @ 616afaa0b956d4f3d3c08adc50b7553e51804d26